components/nimbus/src/tests/stateful/test_nimbus.rs

```diff
@@ -971,3 +971,273 @@ fn test_fetch_enabled() -> Result<()> {
+            "appId": "org.mozilla.fenix",
+            "bucketConfig": {
+                "count": 10000,
+                "start": 0,
+                "total": 10000,
+                "namespace": "secure-gold",
+                "randomizationUnit": "nimbus_id"
+            },
+            "targeting": "true",
+            "userFacingName": "test experiment",
+            "referenceBranch": "control",
+            "isEnrollmentPaused": false,
+            "proposedEnrollment": 7,
+            "userFacingDescription": "This is a test experiment for testing purposes.",
+            "id": "secure-copper",
+            "last_modified": 1_602_197_324_372i64,
+        }
+    ]}))?;
+    client.set_experiments_locally(experiment_json)?;
+    client.apply_pending_experiments()?;
+
+    let active_experiments = client.get_active_experiments()?;
+    assert_eq!(active_experiments.len(), 1);
+
+    let targeting_helper = client.create_targeting_helper(None)?;
+    assert!(targeting_helper.eval_jexl("'test-1' in active_experiments".to_string())?);
+
+    // Apply experiment that targets the above experiment is in enrollments
+    let experiment_json = serde_json::to_string(&json!({
+        "data": [{
+            "schemaVersion": "1.0.0",
+            "slug": "test-2",
+            "endDate": null,
+            "featureIds": ["some-feature-1"],
+            "branches": [
+                {
+                "slug": "control",
+                "ratio": 1
+                },
+                {
+                "slug": "treatment",
+                "ratio": 1
+                }
+            ],
+            "channel": "nightly",
+            "probeSets": [],
+            "startDate": null,
+            "appName": "fenix",
+            "appId": "org.mozilla.fenix",
+            "bucketConfig": {
+                "count": 10000,
+                "start": 0,
+                "total": 10000,
+                "namespace": "secure-gold",
+                "randomizationUnit": "nimbus_id"
+            },
+            "targeting": "'test-1' in enrollments",
+            "userFacingName": "test experiment",
+            "referenceBranch": "control",
+            "isEnrollmentPaused": false,
+            "proposedEnrollment": 7,
+            "userFacingDescription": "This is a test experiment for testing purposes.",
+            "id": "secure-copper",
+            "last_modified": 1_602_197_324_372i64,
+        }
+    ]}))?;
+    client.set_experiments_locally(experiment_json)?;
+    client.apply_pending_experiments()?;
+
+    let active_experiments = client.get_active_experiments()?;
+    assert_eq!(active_experiments.len(), 1);
+
+    let targeting_helper = client.create_targeting_helper(None)?;
+    assert!(!targeting_helper.eval_jexl("'test-1' in active_experiments".to_string())?);
+    assert!(targeting_helper.eval_jexl("'test-2' in active_experiments".to_string())?);
+    assert!(targeting_helper.eval_jexl("'test-1' in enrollments".to_string())?);
```

Comment:
So at this point we've unenrolled from `test-1` and enrolled in `test-2`, so shouldn't both `test-1` and `test-2` appear in `enrollments`?

Reply:
Ah good call I'll add that there too